legal-api/src/legal_api/models/share_structure.py

```diff
@@ -0,0 +1,77 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module holds data for share structure (class and series)."""
+
+from enum import Enum
+
+from sqlalchemy.orm import backref
+
+from .db import db
+
+
+class ShareStructure(db.Model):  # pylint: disable=too-many-instance-attributes
+    """This class manages the share structure (class and series)."""
+
+    class ShareStructureTypes(Enum):
+        """Render an Enum of the share structure types."""
+
+        CLASS = 'class'
+        SERIES = 'series'
+
+    __versioned__ = {}
+    __tablename__ = 'share_structure'
+
+    id = db.Column(db.Integer, primary_key=True)
+    share_type = db.Column('share_type', db.String(20))
```

Comment:
Share class is the parent. so I think that can be added as default

Reply:
Fixed. Based on whether the share structure has a parent or not, the type will be set if it is not set already. Even if the type is set, it will be validated based on the parent id